// Copyright 2023 Intrinsic Innovation LLC
// Intrinsic Proprietary and Confidential
// Provided subject to written agreement between the parties.

syntax = "proto3";

package intrinsic_proto;

// This enum class indicates the limits a joint position command is guaranteeing
// to abide by. For instance,
// `DYNAMIC_LIMITS_CHECK_MODE_CHECK_JOINT_ACCELERATION` denotes it should
// satisfy joint acceleration limits, while
// `DYNAMIC_LIMITS_CHECK_MODE_CHECK_NONE` will not check joint acceleration
// limits. Joint position and velocity constraints are always satisfied.
enum DynamicLimitsCheckMode {
  DYNAMIC_LIMITS_CHECK_MODE_TYPE_UNSPECIFIED = 0;
  DYNAMIC_LIMITS_CHECK_MODE_CHECK_JOINT_ACCELERATION = 1;
  DYNAMIC_LIMITS_CHECK_MODE_CHECK_NONE = 2;
}
